src/inference/Inference.jl

```diff
@@ -535,9 +508,10 @@ end
 floatof(::Type{T}) where {T <: Real} = typeof(one(T)/one(T))
 floatof(::Type) = Real # fallback if type inference failed
 
-@inline Turing.Core.get_matching_type(spl::Turing.Sampler, vi::Turing.RandomVariables.VarInfo, ::Type{<:AbstractFloat}) = floatof(eltype(vi, spl))
-@inline Turing.Core.get_matching_type(spl::Turing.Sampler{<:Hamiltonian}, vi::Turing.RandomVariables.VarInfo, ::Type{Array{T,N}}) where {T, N} = Array{Turing.Core.get_matching_type(spl, vi, T), N}
-@inline Turing.Core.get_matching_type(spl::Turing.Sampler{<:Union{PG, SMC}}, vi::Turing.RandomVariables.VarInfo, ::Type{Array{T,N}}) where {T, N} = TArray{T, N}
+@inline Turing.Core.get_matching_type(spl::AbstractSampler, vi::VarInfo, ::Type{T}) where {T <: AbstractFloat} = floatof(eltype(vi, spl))
```

Comment:
Can probably drop the `@inline`s. Please also re-write to meet 90 char limits for these four methods.